model/validation.go

```diff
@@ -256,46 +258,94 @@ func ValidateMatchCondition(mc *proxyconfig.MatchCondition) (errs error) {
 		errs = multierror.Append(errs, fmt.Errorf("Istio does not support UDP protocol yet"))
 	}
 
-	// TODO We do not (yet) validate http_headers.
+	for name, value := range mc.GetHttpHeaders() {
+		if err := ValidateHTTPHeaderName(name); err != nil {
+			errs = multierror.Append(errs, multierror.Prefix(err, fmt.Sprintf("header name %q invalid: ", name)))
+		}
+		if err := ValidateStringMatch(value); err != nil {
+			errs = multierror.Append(errs, multierror.Prefix(err, fmt.Sprintf("header %q value invalid: ", name)))
+		}
+
+		// validate special `uri` header:
+		// absolute path must be non-empty (https://www.w3.org/Protocols/rfc2616/rfc2616-sec5.html#sec5.1.2)
+		if name == HeaderURI {
+			switch m := value.MatchType.(type) {
```

Comment:
I suppose `Exact` with empty string could be used to match empty headers, but when would it be valid to specify `Prefix` or `Regex` with no corresponding prefix or regex value?